course/page/code_feedback.py

```diff
@@ -135,16 +135,13 @@ def check_list(self, name, ref, data, entry_type=None):
     def check_scalar(self, name, ref, data, accuracy_critical=True,
             rtol=1e-5, atol=1e-8, report_success=True, report_failure=True):
         import numpy as np
+        import sympy as sp
```

Comment:
It should be possible to use `sys.modules` to check if `sympy` was imported, and perform additional checks based on that. Something like this:

```
def check_type(data):
    import numpy as np

    # Built-in and numpy types
    if type(data) in (complex, float, int, np.float32, np.float64, np.int32, np.int64):
        return True

    # If the user imported sympy, also allow sympy types
    # This reuses the module that's already been imported
    if 'sympy' in sys.modules:
        sp = sys.modules['sympy']
        if type(data) in (sp.Number, sp.Float, sp.Integer, sp.Rational, sp.AlgebraicNumber):
            return True

    return False
```